hubverse_annotator/app.py

```diff
@@ -0,0 +1,214 @@
+"""
+A streamlit application for that loads hubverse formatted
+tables and plots model forecasts for the user to compare
+and annotate models.
+
+To run: poetry run streamlit run app.py
+"""
+
+import datetime as dt
+import json
+import logging
+import os
+import time
+
+import altair as alt
+import forecasttools
+import pandas as pd
+import polars as pl
+import streamlit as st
+
+PYRENEW_MODELS = {
+    "CFA_Pyrenew-Pyrenew_HE_COVID": "HE_COVID",
+    "CFA_Pyrenew-Pyrenew_H_COVID": "H_COVID",
+    "CFA_Pyrenew-Pyrenew_HW_COVID": "HW_COVID",
+}
+
+
+def create_forecast_chart(
+    hubverse_table: pl.DataFrame, reference_date: str
+) -> alt.Chart:
+    """
+    Ingests a hubverse table (polars) and a reference_date string,
+    pivots quantiles wide, melts back long, then creates a
+    stacked‐area (streamgraph‐style) chart where X is the discrete
+    set of target_end_date weeks.
+    """
+    wide = hubverse_table.pivot(
+        values="value",
+        index=["model", "target_end_date", "reference_date"],
+        on="output_type_id",
+    ).sort(by=["model", "target_end_date", "reference_date"])
+    pdf = wide.to_pandas().reset_index()
+    pdf["target_end_date"] = pd.to_datetime(pdf["target_end_date"])
+    qcols = [
+        c
+        for c in pdf.columns
+        if isinstance(c, float)
+        or (isinstance(c, str) and c.replace(".", "", 1).isdigit())
+    ]
```

Comment:
Can you pick a more informative variable name?

The logic here seems overly complex. Two options I see:

1. Get the list of quantile columns ahead of time by looking at the relevant column before pivoting.
2. Just see if you can cast a the column name to a float. `c.replace(".", "", 1).isdigit()` seems much more brittle than just `float()`.